### ISSUE TITLE
Fix issues with pointer visibility duration and click delays

### ISSUE THREAD

**Issue by virresh** [virresh/matvt#10]
Point 2 of https://github.com/virresh/matvt/issues/8#issuecomment-812868681
and https://gitter.im/virresh/community?at=60729023ae90f36840733a3d
sums up the issue.

Need to debug and fix timers on posting of runnables.

**Comment by 4388450** [comment]
I might suggest that it will be a great idea if you will add a "long press" enable \ disable option in the app and should only be used when needed. Thanks for your work!

**Comment by virresh** [comment]
@4388450 thanks for the suggestion
In case the project continues, will keep this in mind.

**Comment by 4388450** [comment]
Thanks. Because in the last few hours I had a lot of issues on certain apps because the long click option. So I will really appreciate if you can add a option to disable it

**Comment by KFMDMSolutions** [comment]
the pointer is visible but it doesn't perform the click until you release the button

**Comment by virresh** [comment]
@KFMDMSolutions , do you think it'd be more intuitive if the click is performed as soon as the button is pressed instead of button release?

Any ideas on how to distinguish between a short press and a long press in this case?

**Comment by KFMDMSolutions** [comment]
> @KFMDMSolutions , do you think it'd be more intuitive if the click is performed as soon as the button is pressed instead of button release?
> 
> Any ideas on how to distinguish between a short press and a long press in this case?

lets say i wanna record a 5 min voice message
i have hold the button for 5 min and then speak for 5 min that makes it 10 min to record  a 5min message.

it definably needs some delay to differentiate  between a short press and a long press
but that doesn't mean we need to wait for the click to complete

**Comment by KFMDMSolutions** [comment]
you can start the click after a 2 sec or more delay and stop it onkeyup

**Comment by virresh** [comment]
makes sense, although the clicks sent by the gesture API, don't have that kind of leverage.
It needs a gesture made completely before sending and we can't cancel a gesture after it's sent because it's executed immediately.
This makes implementing things like dragging and holding buttons for voice recording impossible.

Android will need to provide something else altogether in order for us to deal with this issue. This functionality exists in Android for sure (they use it with the inbuilt remotes and touchscreens), but haven't exposed the API for use by third-party apps like mine.

What you're describing will become possible with https://github.com/virresh/matvt/issues/11 
Though probably not implementable at the moment.
